core/capabilities/vault/request_authorizer.go

```diff
@@ -86,72 +92,6 @@ func (r *requestAuthorizer) fetchAllowlistedItem(allowListedRequests []workflow_
 	return nil
 }
 
-func (r *requestAuthorizer) digestForRequest(req jsonrpc.Request[json.RawMessage]) ([32]byte, error) {
-	var seed any
-	switch req.Method {
-	case vaulttypes.MethodSecretsCreate:
-		var createSecretsRequests vaultcommon.CreateSecretsRequest
-		if err := json.Unmarshal(*req.Params, &createSecretsRequests); err != nil {
-			return [32]byte{}, errors.New("error unmarshalling create secrets request: " + err.Error())
-		}
-		seed = vaultcommon.CreateSecretsRequest{
-			EncryptedSecrets: createSecretsRequests.EncryptedSecrets,
-		}
-	case vaulttypes.MethodSecretsUpdate:
-		var updateSecretsRequests vaultcommon.UpdateSecretsRequest
-		if err := json.Unmarshal(*req.Params, &updateSecretsRequests); err != nil {
-			return [32]byte{}, errors.New("error unmarshalling update secrets request: " + err.Error())
-		}
-		seed = vaultcommon.UpdateSecretsRequest{
-			EncryptedSecrets: updateSecretsRequests.EncryptedSecrets,
-		}
-	case vaulttypes.MethodSecretsList:
-		var listSecretsRequests vaultcommon.ListSecretIdentifiersRequest
-		if err := json.Unmarshal(*req.Params, &listSecretsRequests); err != nil {
-			return [32]byte{}, errors.New("error unmarshalling list secrets request: " + err.Error())
-		}
-		seed = vaultcommon.ListSecretIdentifiersRequest{
-			Owner:     listSecretsRequests.Owner,
-			Namespace: listSecretsRequests.Namespace,
-		}
-	case vaulttypes.MethodSecretsDelete:
-		var deleteSecretsRequests vaultcommon.DeleteSecretsRequest
-		if err := json.Unmarshal(*req.Params, &deleteSecretsRequests); err != nil {
-			return [32]byte{}, errors.New("error unmarshalling delete secrets request: " + err.Error())
-		}
-		seed = vaultcommon.DeleteSecretsRequest{
-			Ids: deleteSecretsRequests.Ids,
-		}
-	default:
-		return [32]byte{}, fmt.Errorf("unauthorized method: %s", req.Method)
-	}
-
-	return CalculateRequestDigest(seed), nil
-}
-
-// CalculateRequestDigest creates a SHA256 digest of the request for integrity verification
-// This function is shared between client (JWT generation) and server (JWT validation)
```

Comment:
q: is this something we could combine with the chainlink-common lib? https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/nodeauth/jwt/node_jwt_generator.go#L31

Or is this a completely different flow? (i.e. not based on a JWT self-signed by a CSA key)